.githooks/pre-commit

```diff
@@ -42,14 +42,29 @@ check_docker_vars "MAILER_EMAIL_SMTP_URL"
 check_docker_vars "MAILER_EMAIL_SMTP_PORT"
 check_docker_vars "WEBSITE_HOST"
 
-## Checking for custom settings in sqe-http-api/appsettings.json
-echo "Checking for private data in sqe-http-api/appsettings.json"
+## Checking for custom settings in sqe-api-server/appsettings.json
+echo "Checking for private data in sqe-api-server/appsettings.json"
 check_appsettings_vars "MailerEmailAddress"
 check_appsettings_vars "MailerEmailUsername"
 check_appsettings_vars "MailerEmailPassword"
 check_appsettings_vars "MailerEmailSmtpUrl"
 check_appsettings_vars "MailerEmailSmtpPort"
 
+## We autogenerate the Hub methods from the Controllers
+function generate_hubs
+{
+    dotnet run --project sqe-realtime-hub-builder/sqe-realtime-hub-builder.csproj
+    if [[ "$?" != 0 ]]; then 
+        printf "\e[31mThere are errors in the HTTP controllers in 'sqe-api-server/HttpControllers'.\n"
+        printf "Please fix these errors before continuing to commit your changes.\e[0m\n"
+        exit 1
+    fi
+}
+
+## Generating Hub Methods
+echo "Autogenerating realtime Hub methods"
+generate_hubs
+
```

Comment:
I like the idea of these checks. I hope they won't cause issues with the ordinary workflow. One thing I think of - we try to push whatever it is we have (even when it's not working) when we go home - for backup reasons. We do it in branches that we know don't bother anybody else. Perhaps we should have some naming convention for branches that aren't checked, just for this workflow (it beats setting up a backup service on each and every computer we have)